src/hooks/fetch.ts

```diff
@@ -335,31 +335,43 @@ export class FetchInstrumentation {
   /**
    * Converts the headers object to a key-value pair object.
    * Fetch library uses multiple format to represent headers, this function will convert them all to a key-value pair object.
-   * @param {[string, string][] | Record<string, string> | Headers} headers Headers object as used by the fetch library
+   * @param {string[][] | Record<string, string | ReadonlyArray<string>> | Headers} headers Headers object as used by the fetch library
    * @returns {Record<string, string>} The headers as a key-value pair object
    * @private
    */
   private static convertHeadersToKeyValuePairs(
     // @ts-ignore
-    headers: [string, string][] | Record<string, string> | Headers
+    headers: string[][] | Record<string, string | ReadonlyArray<string>> | Headers
   ): Record<string, string> {
+    const headersObject: Record<string, string> = {};
+
+    // Handle Headers object
     // @ts-ignore
     if (headers instanceof Headers) {
-      const headersObject: Record<string, string> = {};
       headers.forEach((value, key) => {
         headersObject[key] = value;
       });
-      return headersObject;
     }
-    if (Array.isArray(headers)) {
-      const headersObject: Record<string, string> = {};
+    // Handle string[][] (array of key-value pairs)
+    else if (Array.isArray(headers)) {
       headers.forEach(([key, value]) => {
         headersObject[key] = value;
       });
-      return headersObject;
+    }
+    // Handle Record<string, string | ReadonlyArray<string>>
+    else if (headers && typeof headers === 'object' && !Array.isArray(headers)) {
```

Comment:
the `!Array.isArray(headers)` addition doesn't do anything, because it comes inside the else if block after the previous check, so we reach this check only if the headers var isn't an array

Reply:
Done